qiime/core/testing/type.py

```diff
@@ -0,0 +1,15 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import qiime.plugin
+
+
+IntSequence1 = qiime.plugin.SemanticType('IntSequence1')
+IntSequence2 = qiime.plugin.SemanticType('IntSequence2')
+Mapping = qiime.plugin.SemanticType('Mapping')
+FourInts = qiime.plugin.SemanticType('FourInts')
```

Comment:
There isn't a whole lot of type-system integration tests yet (e.g. exercising `TypeMap`). Okay to add more type-related tests in the future? There's many many many more test cases to be added but I want to get this merged ASAP so we have a reasonable testbase to work from (and have master back to passing again).